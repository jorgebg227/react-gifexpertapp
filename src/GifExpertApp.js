import React, {useState} from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({defaultCategories = []})=>{
    const [categories, setCategories] = useState(defaultCategories);

    /*
    const handleAdd = (e) => {
        const anadir = 'HunterXHunter'
        //setCategories([...categories], anadir);  // a単ade al final
        //setCategories([anadir, ...categories]); // a単ade al inicio
        setCategories(cats => [...cats, anadir]); // a単ade al final
        //setCategories(cats => [anadir, ...cats]); // a単ade al Inicio
    }
    */

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr />
            
            <ol>
                {
                    categories.map((category) => (
                        <GifGrid key={category} category={category} />
                    ))
                }
            </ol>
        </>
    );
}

export default GifExpertApp;