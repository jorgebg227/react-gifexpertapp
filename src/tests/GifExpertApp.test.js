import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas con <GifExpertApp />', () => {
    const wrapper = shallow(<GifExpertApp />);
    
    test('Debe de mostrar correctamente <GifExpertApp />', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('debe de mostrar una lista de categorias', ()=>{
        const categories =['Nirvana', 'Real Madrid'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
});
