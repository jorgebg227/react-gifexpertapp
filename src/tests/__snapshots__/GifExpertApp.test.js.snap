// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <GifExpertApp /> Debe de mostrar correctamente <GifExpertApp /> 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas con <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Nirvana"
      key="Nirvana"
    />
    <GifGrid
      category="Real Madrid"
      key="Real Madrid"
    />
  </ol>
</Fragment>
`;
